frontend/src/tbpro/elements/SwitchToggle.vue

```diff
@@ -0,0 +1,107 @@
+<script setup lang="ts">
+import { ref, onMounted } from 'vue';
+import { useI18n } from 'vue-i18n';
+
+// component constants
+const { t } = useI18n();
+
+// component emits
+const emit = defineEmits(['changed']);
+
+// component properties
+interface Props {
+  active: boolean; // initial toggle state
```

Comment:
Can you add name so we can use form elements? Thanks! 

(I know we don't use them right now except for login and contact, but it might come in handy when developing traditional forms)

Reply:
Of course, good thinking!